### ISSUE TITLE
Broken mir with variadic static function pointer

### ISSUE THREAD

**Issue by panstromek** [rust-lang/rust#74655]
### Code

```Rust
static mut foo: Option<unsafe extern "C" fn(_, ...)> = None;

fn main() {}
```

### Meta

`rustc --version --verbose`:
```
ustc 1.46.0-nightly (346aec9b0 2020-07-11)
binary: rustc
commit-hash: 346aec9b02f3c74f3fce97fd6bda24709d220e49
commit-date: 2020-07-11
host: x86_64-unknown-linux-gnu
release: 1.46.0-nightly
LLVM version: 10.0

```

### Error output

```
error: internal compiler error: bad_placeholder_type
 --> min-ice/src/main.rs:1:45
  |
1 | static mut foo: Option<unsafe extern "C" fn(_, ...)> = None;
  |                                             ^
  |
  = note: delayed at /rustc/346aec9b02f3c74f3fce97fd6bda24709d220e49/src/librustc_session/session.rs:436:27

error: internal compiler error: TyKind::Error constructed but no error reported
  |
  = note: delayed at /rustc/346aec9b02f3c74f3fce97fd6bda24709d220e49/src/librustc_session/session.rs:436:27

error: internal compiler error: cat_expr Errd
 --> min-ice/src/main.rs:1:56
  |
1 | static mut foo: Option<unsafe extern "C" fn(_, ...)> = None;
  |                                                        ^^^^
  |
  = note: delayed at /rustc/346aec9b02f3c74f3fce97fd6bda24709d220e49/src/librustc_session/session.rs:436:27

error: internal compiler error: mir_const_qualif: MIR had errors
 --> min-ice/src/main.rs:1:1
  |
1 | static mut foo: Option<unsafe extern "C" fn(_, ...)> = None;
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: delayed at /rustc/346aec9b02f3c74f3fce97fd6bda24709d220e49/src/librustc_session/session.rs:436:27

error: internal compiler error: PromoteTemps: MIR had errors
 --> min-ice/src/main.rs:1:1
  |
1 | static mut foo: Option<unsafe extern "C" fn(_, ...)> = None;
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: delayed at /rustc/346aec9b02f3c74f3fce97fd6bda24709d220e49/src/librustc_session/session.rs:436:27

error: internal compiler error: broken MIR in DefId(0:3 ~ min_ice[2aeb]::foo[0]) ("return type"): bad type [type error]
 --> min-ice/src/main.rs:1:1
  |
1 | static mut foo: Option<unsafe extern "C" fn(_, ...)> = None;
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: delayed at src/librustc_mir/borrow_check/type_check/mod.rs:258:27

error: internal compiler error: broken MIR in DefId(0:3 ~ min_ice[2aeb]::foo[0]) (LocalDecl { mutability: Mut, local_info: None, internal: false, is_block_tail: None, ty: [type error], user_ty: None, source_info: SourceInfo { span: min-ice/src/main.rs:1:1: 1:61 (#0), scope: scope[0] } }): bad type [type error]
 --> min-ice/src/main.rs:1:1
  |
1 | static mut foo: Option<unsafe extern "C" fn(_, ...)> = None;
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: delayed at src/librustc_mir/borrow_check/type_check/mod.rs:258:27


```

<details><summary><strong>Backtrace</strong></summary>
<p>

```
thread 'rustc' panicked at 'no errors encountered even though `delay_span_bug` issued', src/librustc_errors/lib.rs:367:17
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/libunwind.rs:86
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.46/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print_fmt
             at src/libstd/sys_common/backtrace.rs:78
   3: <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt
             at src/libstd/sys_common/backtrace.rs:59
   4: core::fmt::write
             at src/libcore/fmt/mod.rs:1076
   5: std::io::Write::write_fmt
             at src/libstd/io/mod.rs:1537
   6: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:62
   7: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:49
   8: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:198
   9: std::panicking::default_hook
             at src/libstd/panicking.rs:217
  10: rustc_driver::report_ice
  11: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:530
  12: std::panicking::begin_panic
  13: <rustc_errors::HandlerInner as core::ops::drop::Drop>::drop
  14: core::ptr::drop_in_place
  15: <alloc::rc::Rc<T> as core::ops::drop::Drop>::drop
  16: core::ptr::drop_in_place
  17: rustc_span::with_source_map
  18: rustc_interface::interface::create_compiler_and_run
  19: scoped_tls::ScopedKey<T>::set
```

</p>
</details>

**Comment by panstromek** [comment]
I also reproduced it on stable.

**Comment by jonas-schievink** [comment]
This is fixed on the current nightly.
